src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -0,0 +1,1560 @@
+        string partitionId = rowData.at(6).second;
+
+        if ((ip.find("localhost") != std::string::npos) || ip == masterIP) {
+            host = ip;
+        } else {
+            host = user + "@" + ip;;
+        }
+
+        if( partitionMap.find(workerID) == partitionMap.end()){
+            std::vector<string> partitionVec;
+            partitionVec.push_back(partitionId);
+            partitionMap.insert(std::pair<string, std::vector<string>>(workerID, partitionVec));
+        } else {
+            std::vector<string> partitionVec = partitionMap.find(workerID)->second;
+            partitionVec.push_back(partitionId);
+        }
+
+        triangleCount_logger.log("###TRIANGLE-COUNT-EXECUTOR### Getting Triangle Count : Host " + host + " Server Port " +
+                            serverPort + " PartitionId " + partitionId, "info");
+
+    }
+
+    for (auto &&futureCall:remoteCopyRes) {
+        futureCall.wait();
+    }
+
+    for (int i = 0; i < workerListSize; i++) {
+        Utils::worker currentWorker = workerList.at(i);
+        string host = currentWorker.hostname;
+        string workerID = currentWorker.workerID;
+        string partitionId;
+
+        std::vector<string> partitionList = partitionMap[workerID];
+
+        std::vector<string>::iterator partitionIterator;
+
+        for (partitionIterator = partitionList.begin(); partitionIterator != partitionList.end(); ++partitionIterator) {
+            partitionCount++;
+            int workerPort = atoi(string(currentWorker.port).c_str());
+            int workerDataPort = atoi(string(currentWorker.dataPort).c_str());
+            triangleCount_logger.log("====>PORT:" + std::to_string(workerPort), "info");
+
+            partitionId = *partitionIterator;
+            intermRes.push_back(
+                    std::async(std::launch::async, TriangleCountExecutor::getTriangleCount, atoi(graphId.c_str()), host,
+                               workerPort, workerDataPort, atoi(partitionId.c_str()), masterIP, uniqueId,
+                               isCompositeAggregation, threadPriority));
+        }
+    }
+
+    PerformanceUtil performanceUtil;
+    performanceUtil.init();
+
+    std::string query = "SELECT attempt from graph_sla INNER JOIN sla_category where graph_sla.id_sla_category=sla_category.id and "
+                        "graph_sla.graph_id='" + graphId + "' and graph_sla.partition_count='" + std::to_string(partitionCount) +
+                        "' and sla_category.category='" + Conts::SLA_CATEGORY::LATENCY + "';";
+
+    std::vector<vector<pair<string, string>>> queryResults = perfDB.runSelect(query);
+
+    if (queryResults.size() > 0) {
+        std::string attemptString = queryResults[0][0].second;
+        int calibratedAttempts = atoi(attemptString.c_str());
+
+        if (calibratedAttempts >= Conts::MAX_SLA_CALIBRATE_ATTEMPTS) {
+            canCalibrate = "false";
+        }
+    }
+
+    while (!workerResponded && canCalibrate == "true") {
```

Comment:
We should change this to `canCalibrate.compare("true") == 0`

Reply:
Changed this parameter to boolean